systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java

```diff
@@ -33,107 +41,108 @@
 
-            String zkSsName = KafkaResources.zookeeperStatefulSetName("my-cluster");
-            String kafkaSsName = KafkaResources.kafkaStatefulSetName("my-cluster");
-            String eoDepName = KafkaResources.entityOperatorDeploymentName("my-cluster");
-            Map<String, String> zkPods = StUtils.ssSnapshot(CLIENT, NAMESPACE, zkSsName);
-            Map<String, String> kafkaPods = StUtils.ssSnapshot(CLIENT, NAMESPACE, kafkaSsName);
-            Map<String, String> eoPods = StUtils.depSnapshot(CLIENT, NAMESPACE, eoDepName);
+            makeSnapshots();
 
             List<Pod> pods = CLIENT.pods().inNamespace(NAMESPACE).withLabels(CLIENT.apps().statefulSets().inNamespace(NAMESPACE).withName(zkSsName).get().getSpec().getSelector().getMatchLabels()).list().getItems();
             for (Pod pod : pods) {
                 LOGGER.info("Pod {} has image {}", pod.getMetadata().getName(), pod.getSpec().getContainers().get(0).getImage());
             }
 
-            // Upgrade the CO, to current HEAD,
-            LOGGER.info("Updating");
-            copyModifyApply(new File("../install/cluster-operator"));
-            LOGGER.info("Waiting for CO redeployment");
-            KUBE_CLIENT.waitForDeployment("strimzi-cluster-operator", 1);
-
-            LOGGER.info("Waiting for ZK SS roll");
-            StUtils.waitTillSsHasRolled(CLIENT, NAMESPACE, zkSsName, zkPods);
-            LOGGER.info("Checking ZK pods using new image");
-            waitTillAllPodsUseImage(CLIENT.apps().statefulSets().inNamespace(NAMESPACE).withName(zkSsName).get().getSpec().getSelector().getMatchLabels(),
-                    "strimzi/zookeeper:latest-kafka-2.0.0");
-            LOGGER.info("Waiting for Kafka SS roll");
-            StUtils.waitTillSsHasRolled(CLIENT, NAMESPACE, kafkaSsName, kafkaPods);
-            LOGGER.info("Checking Kafka pods using new image");
-            waitTillAllPodsUseImage(CLIENT.apps().statefulSets().inNamespace(NAMESPACE).withName(kafkaSsName).get().getSpec().getSelector().getMatchLabels(),
-                    "strimzi/kafka:latest-kafka-2.1.0");
-            LOGGER.info("Waiting for EO Dep roll");
-            // Check the TO and UO also got upgraded
-            StUtils.waitTillDepHasRolled(CLIENT, NAMESPACE, eoDepName, eoPods);
-            LOGGER.info("Checking EO pod using new image");
-            waitTillAllContainersUseImage(
-                    CLIENT.apps().deployments().inNamespace(NAMESPACE).withName(eoDepName).get().getSpec().getSelector().getMatchLabels(),
-                    0,
-                    "strimzi/topic-operator:latest");
-            waitTillAllContainersUseImage(
-                    CLIENT.apps().deployments().inNamespace(NAMESPACE).withName(eoDepName).get().getSpec().getSelector().getMatchLabels(),
-                    1,
-                    "strimzi/user-operator:latest");
+            // Execution of required procedures before upgrading CO
+            if (!procedures.isEmpty()) {
```

Comment:
We are just adding Kafka 2.2.0 and discussing removing Kafka 2.0.x ... I think this cannot work long term with hardcoded Kafka versions.

Reply:
Yes, but we still need Kafka 2.0.x for cases when we want to upgrade Strimzi from major version 0.8.x to 0.x.x. SO this is a reason why we still should have it in ST